src/lib.rs

```diff
@@ -182,6 +182,21 @@ impl PublicKey {
     pub fn to_bytes(self) -> [u8; PK_SIZE] {
         self.0.to_compressed()
     }
+
+    /// Deserialize a hex-encoded representation of a `PublicKey` to a `PublicKey` instance.
+    pub fn from_hex(hex: &str) -> Result<Self> {
+        let pk_bytes = hex::decode(hex).map_err(Error::HexConversionFailed)?;
```

Comment:
Now that you used the #[from] macro, you can just use `?` and the error will be converted into the `HexConversionFailed` error variant.
```suggestion
        let pk_bytes = hex::decode(hex)?;
```